### DISCUSSION TITLE
[mix_time_series]Adding annotation to chart

### DISCUSSION THREAD

**Discussion by agatapst** [apache/superset#18421] [Ideas]
In dual line chart, which according to [superset-ui](https://github.com/apache-superset/superset-uil) does not have any supported annotation types, it is possible to open modal with adding new annotation. Bar chart does not have any supported annotation types and this section is hidden - it works fine in that case.

In [superset-ui](https://github.com/apache-superset/superset-uil) `controlPanel` for `DualLine`, `sections.annotations` should be removed. That way it will be consistent with `DualLineChartPlugin` metadata, that Dual Line chart does not have any supported annotation types.

It is worth checking if it is consistent for other charts to avoid future bugs. 
### Expected results
In the left bar, Add Annotation Layer button and the whole annotation section should be hidden. 

### Actual results
The annotation section is displayed and the modal opens. It is impossible to add any working annotation.

#### Screenshots
![annotations_dual_line_before](https://user-images.githubusercontent.com/47450693/100920371-4ed50e80-34db-11eb-9204-2e0626e8161a.gif)

#### How to reproduce the bug

1. Go to Explore
2. Create dual line chart
3. Click Add Annotation Layer in the left bat

### Environment

- superset version: current master (commit 44e80e)
- python version: 3.7.9
- node.js version: 14.15.1

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context

Partly connected with PR #11876

**Comment by junlincc** [comment]
this is awesome!thank you Agata for point out issues as you move. 🙏 i removed the Polidea label for now as this issue may not land on you guys.
